datafusion/sql/src/expr/function.rs

```diff
@@ -70,6 +70,50 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             return Ok(Expr::ScalarFunction(ScalarFunction::new_udf(fm, args)));
         }
 
+        // Build Unnest expression
+        if name.eq("unnest") {
+            let exprs =
+                self.function_args_to_expr(args.clone(), schema, planner_context)?;
+
+            match exprs.len() {
+                0 => {
+                    return exec_err!("unnest() requires at least one argument");
+                }
+                1 => {
+                    if let Expr::ScalarFunction(ScalarFunction {
+                        func_def:
+                            ScalarFunctionDefinition::BuiltIn(
+                                BuiltinScalarFunction::MakeArray,
+                            ),
+                        ..
+                    }) = exprs[0]
+                    {
+                        // valid
+                    } else if let Expr::Column(_) = exprs[0] {
+                        // valid
+                    } else if let Expr::ScalarFunction(ScalarFunction {
+                        func_def:
+                            ScalarFunctionDefinition::BuiltIn(BuiltinScalarFunction::Struct),
+                        ..
+                    }) = exprs[0]
+                    {
+                        return not_impl_err!("unnest() does not support struct yet");
+                    } else {
+                        return plan_err!(
+                            "unnest() can only be applied to array and structs and null"
```

Comment:
this line says unnest supports struct but line above says the opposite

Reply:
first line is struct nyi error.
second line is to tell you that only those 3 are valid.

Reply:
thats the confusing part. `not_impl_err!("unnest() does not support struct yet");` - here the user can get the struct is not supported by unnest.
```plan_err!("unnest() can only be applied to array and structs and null") ``` here the message says that unnest support ONLY array, structs and nulls which confronts with first err message

Maybe I'm missing something?